data_model/src/events/time.rs

```diff
@@ -0,0 +1,110 @@
+//! Time event and filter
+
+use core::{ops::Range, time::Duration};
+
+use super::*;
+
+/// Special event that is emitted when `WSV` is ready for handling time-triggers
+///
+/// Contains time interval which is used to identify time-triggers to be executed
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Decode, Encode, IntoSchema)]
+pub struct Event(pub Interval);
+
+/// Filters time-events and allows only ones which time interval contains
+#[derive(
+    Debug,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Clone,
+    Copy,
+    Decode,
+    Encode,
+    IntoSchema,
+    Hash,
+    Serialize,
+    Deserialize,
+)]
+pub struct EventFilter(pub Occurrence);
+
+impl EventFilter {
+    /// Check if `event` matches filter
+    ///
+    /// Returns `true`, if `event` matches filter and `false` if not
+    pub fn matches(&self, event: &Event) -> bool {
+        match &self.0 {
+            Occurrence::Every(interval) => {
+                let time = interval.since + interval.length;
+                Range::from(event.0).contains(&time)
+            }
+            Occurrence::ExactlyAt(time) => Range::from(event.0).contains(time),
+        }
+    }
+}
+
+/// Enumeration of possible occurrence schemes
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Encode,
+    Decode,
+    Serialize,
+    Deserialize,
+    IntoSchema,
+    Hash,
+)]
+pub enum Occurrence {
+    /// Occurs every set time
+    Every(Interval),
```

Comment:
Why not `Option<Duration>`?

Reply:
Do you mean `Every(Option<Duration>)`? Then what `Every(None)` should mean?

Also I have to store last execution time (initally registration time) somewhere

Reply:
I've added some new commits. If there is still a place for this remark, please, let me know. Cause I don't know how to implement this suggestion now

Reply:
I think, Sato suggested the same thing here: https://github.com/hyperledger/iroha/pull/1961#discussion_r829691512
So I'll close this conversation